File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        freq=[0]*26
        left=0
        maxfreq=0
        ans=0
        for right in range(0,len(s)):
            freq[ord(s[right])-ord('A')]+=1;
            maxfreq=max(maxfreq,freq[ord(s[right])-ord('A')])
            while ((right-left+1) - maxfreq) > k:
                freq[ord(s[left])-ord('A')]-=1
                left+=1
            ans=max(ans,right-left+1)
        return ans



        

File: Data Structures & Algorithms/palindrome-partitioning/submission-1.java
class Solution 
{
    List<List<String>> res = new ArrayList<>();
    public boolean ispal(String s)
    {
        int left=0;
        int right=s.length()-1;
        while(left<right)
        {
            if(s.charAt(left)!=s.charAt(right))
            {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
    public void build(String s, List<String> ans,int i)
    {
        if(i==s.length())
        {
            res.add(new ArrayList<>(ans));
            return;
        }
        for(int end = i;end<s.length();end++)
        {
            String part = s.substring(i, end + 1);
            if(ispal(part))
            {
                 ans.add(s.substring(i, end + 1));
                build(s,ans,end+1);
                ans.remove(ans.size()-1);
            }
        }   
    }
    public List<List<String>> partition(String s) 
    {
        List<String> ans = new ArrayList<>();
        build(s,ans,0);
        return res;
    }
}
